### DISCUSSION TITLE
PublishSingleFile 方式发布程序问题

### DISCUSSION THREAD

**Discussion by shuxinqin** [dotnetcore/Natasha#103] [Q&A]
如果使用 PublishSingleFile 发布程序，貌似 Natasha 不能用了，提示引用的依赖不存在，不知是我打开姿势不对还是怎么回事？
dotnet publish -r win-x64 /p:PublishSingleFile=true
https://docs.microsoft.com/en-us/dotnet/core/deploying/single-file

**Comment by NMSAzulX** [comment]
DLL裁剪 和 动态编译理论上是冲突的,  裁剪会尽可能小, 只携带有用的程序集, 动态编译需要元数据和程序集的支持,  尽可能打包多的 dll.   Natasha 支持按需加载引用, 你可以使用裁剪, 但编译所需的dll 你也要清楚,  使用按需引用Api  `domain.AddReferecemXX` 即可.
